loky/backend/_thread_pool_limiters.py

```diff
@@ -0,0 +1,481 @@
+
+#############################################################################
+# The following provides utilities to load C-libraries that relies on thread
+# pools and limit the maximal number of thread that can be used.
+#
+#
+import os
+import sys
+import ctypes
+import threading
+from ctypes.util import find_library
+
+from ..backend.context import cpu_count
+
+
+if sys.platform == "darwin":
+
+    # On OSX, we can get a runtime error due to multiple OpenMP libraries
+    # loaded simultaneously. This can happen for instance when calling BLAS
+    # inside a prange. Setting the following environment variable allows
+    # multiple OpenMP libraries to be loaded. It should not degrade
+    # performances since we manually take care of potential over-subscription
+    # performance issues, in sections of the code where nested OpenMP loops can
+    # happen, by dynamically reconfiguring the inner OpenMP runtime to
+    # temporarily disable it while under the scope of the outer OpenMP parallel
+    # section.
+    os.environ.setdefault("KMP_DUPLICATE_LIB_OK", "True")
+
+
+def openmp_num_threads(n_threads=None):
+    """Determine the effective number of threads used for parallel OpenMP calls
+
+    For n_threads=None, returns openmp.omp_get_max_threads().
+    For n_threads > 0, use this as the maximal number of threads for parallel
+    OpenMP calls and for n_threads < 0, use the maximal number of threads minus
+    - n_threads + 1.
+    Raise a ValueError for n_threads=0.
+    """
+    if n_threads == 0:
+        raise ValueError("n_threads=0 is invalid")
+    elif n_threads < 0:
+        return max(1, cpu_count() + n_threads + 1)
```

Comment:
I am not sure we should not rely on `loky.cpu_count` here. If the user has passed `OMP_NUM_THREADS` environment variable it would be ignored. Why not just use `openmp.omp_get_max_threads()` as stated in the docstring?

Reply:
Because we do not actually need `openmp_num_threads` in for the thread pool limit logic, I think we should completely remove this function from loky.

Projects that need this kind of negative num thread logic like scikit-learn will implement it them-self instead.

Reply:
ok I agree